peps/pep-0735.rst

```diff
@@ -0,0 +1,476 @@
+Group as a subtable, thus enabling us to attach future keys to each group,
+allows for the greatest future flexibility.
+
+However, it also makes the structure nested more deeply, and therefore harder
+to teach and learn. One of the goals of this PEP is to be an easy replacement
+for many ``requirements.txt`` use-cases.
+
+Why not restrict dependencies to PEP 508 only?
+----------------------------------------------
+
+There are valid use-cases for
+
+* including one dependency group in another
+* including the current package (if the project is a package)
+* including the current package with extras (if the project is a package)
+
+These are not satisfiable without some expansion of syntax beyond what is
+possible with PEP 508.
+
+Why not define keys in dependency specifications for common options seen in ``requirements.txt`` (e.g. ``--hash``)?
+-------------------------------------------------------------------------------------------------------------------
+
+It is currently unclear which options will be the most necessary and beneficial.
+
+Certain problems, e.g. package hashing, are the domain of lockfiles.
+The data in this PEP is meant to be lockfile *input*, not necessarily lockfile *output*.
+Therefore, hashing should not be privileged, nor should other options.
+
+The PEP defines space for future expansion of the data format and mandates that
+tools support and parse it such that expansion will be a non-breaking change.
+
+Why not restrict dependency specifications to strings only?
+-----------------------------------------------------------
+
+Failing to establish the object format at this stage would lead to a breaking
+change if it were ever introduced.
+Rather than having a smooth degradation path, users would experience breakage
+if the object format were introduced and only some tools supported it.
+
+Why is the table not named ``[run]``, ``[dependency_groups]``, ...?
+-------------------------------------------------------------------
+
+There are many possible names for this concept.
+It will have to live alongside the already existing ``[project.dependencies]``
+and ``[project.optional-dependencies]`` tables, and possibly a new
+``[external]`` dependency table as well (at time of writing, PEP 725 is in
+progress).
+
+``[run]`` was a leading proposal in earlier discussions, but its proposed usage
+centered around a single set of runtime dependencies. This PEP explicitly
+outlines multiple groups of dependencies, which makes ``[run]`` a less
+appropriate fit -- this is not just dependency data for a specific runtime
+context, but for multiple contexts.
+
+``[dependency_groups]`` is a reasonable name, but it fails to namespace the
+dependency group data under a related umbrella term. As a result, it would be
+harder to extend in the future to include other data, such as python version
+requirements.
+
+.. note::
+
+    No name has been declared final yet.
```

Comment:
This would make the topic an open issue then.

Reply:
Ah! I can make that update. I got confused about what that section was for (in the context of potential GitHub Issues).